File: backend/src/main/java/com/movie/data/Movie.java
package com.movie.data;

import java.util.Date;
import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Data
public class Movie {
    String plot;
    String title;
    List<String> cast;
    List<String> genres;
    List<String> languages;
    String rated;
    Date released;
    Imdb imdb;
    List<String> countries;
    String poster;
}

File: frontend/src/data/movie/movie.ts
import { z } from "zod";

export const movieSchema = z.object({
    plot: z.string(),
    title: z.string(),
    cast: z.string().array(),
    genres: z.string().array(),
    languages: z.string().array(),
    rated: z.string(),
    released: z.date(),
    imdb: z.object({
        id: z.number(),
        votes: z.number(),
        rating: z.number()
    }),
    countries: z.string().array(),
    poster: z.string().url()
});

export type Movie = z.infer<typeof movieSchema>;